src/data/list/infix.lean

```diff
@@ -307,6 +331,27 @@ begin
   exact infix_append _ _ _
 end
 
+instance : is_refl (list α) (<+:) := ⟨prefix_refl⟩
+instance : is_trans (list α) (<+:) := ⟨λ _ _ _, is_prefix.trans⟩
+instance : is_antisymm (list α) (<+:) := ⟨λ l₁ l₂ h₁ h₂,
+  eq_of_prefix_of_length_eq h₁ (le_antisymm (is_prefix.length_le h₁) (is_prefix.length_le h₂))⟩
+instance : is_preorder (list α) (<+:) := {}
+instance : is_partial_order (list α) (<+:) := {}
```

Comment:
Better to just provide these all at once:
```suggestion
instance : is_partial_order (list α) (<+:) :=
{ refl := λ _, prefix_rfl,
  trans := λ _ _ _, is_prefix.trans,
  antisymm := λ l₁ l₂ h₁ h₂, eq_of_prefix_of_length_eq h₁ $ h₁.length_le.antisymm h₂.length_le }
```
Lean knows how to find all the other instances from this one